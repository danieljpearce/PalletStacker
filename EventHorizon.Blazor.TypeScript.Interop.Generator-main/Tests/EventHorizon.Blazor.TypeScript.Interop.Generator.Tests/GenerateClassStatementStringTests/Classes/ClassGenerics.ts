declare module Examples {
    export class Camera { }
    export class ExampleClassWithGeneric<TCamera> { }

    export class ExampleClass extends ExampleClassWithGeneric<Camera> {
        /**
         * Instantiates a new ArcRotateCameraInputsManager.
         * @param camera Defines the camera the inputs belong to
         */
        constructor(camera: Camera);
        /**
         * Add mouse wheel input support to the input manager.
         * @returns the current input manager
         */
        addMouseWheel(): Camera;
        /**
         * Add pointers input support to the input manager.
         * @returns the current input manager
         */
        addPointers(): Camera;
        /**
         * Add keyboard input support to the input manager.
         * @returns the current input manager
         */
        addKeyboard(): Camera;
    }
}