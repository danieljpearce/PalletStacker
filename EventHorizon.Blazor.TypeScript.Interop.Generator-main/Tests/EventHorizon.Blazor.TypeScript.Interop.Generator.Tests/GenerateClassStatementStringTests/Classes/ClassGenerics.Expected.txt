/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : ExampleClassWithGeneric<Camera>
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
        }

        public ExampleClass(
            Camera camera
        ) : base()
        {
            var entity = EventHorizonBlazorInterop.New(
                new string[] { "Examples", "ExampleClass" },
                camera
            );
            ___guid = entity.___guid;
        }
        #endregion

        #region Methods
        public Camera addMouseWheel()
        {
            return EventHorizonBlazorInterop.FuncClass<Camera>(
                entity => new Camera() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { this.___guid, "addMouseWheel" }
                }
            );
        }

        public Camera addPointers()
        {
            return EventHorizonBlazorInterop.FuncClass<Camera>(
                entity => new Camera() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { this.___guid, "addPointers" }
                }
            );
        }

        public Camera addKeyboard()
        {
            return EventHorizonBlazorInterop.FuncClass<Camera>(
                entity => new Camera() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { this.___guid, "addKeyboard" }
                }
            );
        }
        #endregion
    }
}